RFC update text (#101)
mwelzl/draft-ccwg-ratelimited-increase: draft-ietf-ccwg-ratelimited-increase.md

This document updates {{Section 5.1 of !RFC4341}} by adding the text in {{rules}} to specify how the cwnd is managed when the sender is rate-limited.

## RFC 5681: TCP Congestion Control

{{!RFC5681}} specifies no limit on cwnd growth in the standard TCP behavior
when a TCP sender is unable to send at the maximum rate allowed by the cwnd.

This document updates {{!RFC5681}} by adding the text in {{rules}} to specify how the cwnd is managed when the sender is rate-limited.

## RFC 9002: QUIC Loss Detection and Congestion Control

{{Section 7.8 of !RFC9002}} states:

>"When bytes in flight is smaller than the congestion window and sending is not pacing limited, the congestion window is underutilized. This can happen due to insufficient application data or flow control limits. When this occurs, the congestion window SHOULD NOT be increased in either slow start or congestion avoidance."

This limits the cwnd growth in accordance with Rate-Limited Increase, but it is more conservative.

{{!RFC9002}} is updated by this document by replacing the final sentence in the cited text with the text in ({{rules}}) to specify the management of the

> cwnd when the sender is rate-limited.

Author: ```suggestion
```

# Updates to RFCs 4341, 5681, 9002, 9260, and 9438 {#rfc-updates}

## RFC 4341: The Datagram Congestion Control Protocol (DCCP) Congestion Control ID (CCID) 2

According to {{!RFC4341}}, a DCCP CCID specifying TCP-like behavior is allowed to grow the cwnd without limit during an uncongested period when it sends at a rate unconstrained by the current cwnd.
This document updates {{Section 5.1 of !RFC4341}} by adding the text in {{rules}} to specify how the cwnd is managed when the sender is rate-limited.

## RFC 5681: TCP Congestion Control

{{!RFC5681}} specifies no limit on cwnd growth in the standard TCP behavior
when a TCP sender is unable to send at the maximum rate allowed by the cwnd.

{{!RFC5681}} is updated by this document by adding the text in ({{rules}}) to specify the management of the

> cwnd when the sender is rate-limited.

Author: ```suggestion
```

# Updates to RFCs 4341, 5681, 9002, 9260, and 9438 {#rfc-updates}

## RFC 4341: The Datagram Congestion Control Protocol (DCCP) Congestion Control ID (CCID) 2

> A DCCP CCID specifying TCP-like behaviour ought to follow the method specified in this document.

Reviewer: ```suggestion
According to {{!RFC4341}, a DCCP CCID specifying TCP-like behaviour is allowed to grow an unlimited cwnd during an uncongested period when it sends at a rate unconstrained by the current cwnd. 
```

> According to {{!RFC4341}, a DCCP CCID specifying TCP-like behaviour is allowed to grow an unlimited cwnd during an uncongested period when it sends at a rate unconstrained by the current cwnd. 

Author: ```suggestion
According to {{!RFC4341}}, a DCCP CCID specifying TCP-like behaviour is allowed to grow an unlimited cwnd during an uncongested period when it sends at a rate unconstrained by the current cwnd. 
```

> According to {{!RFC4341}}, a DCCP CCID specifying TCP-like behaviour is allowed to grow an unlimited cwnd during an uncongested period when it sends at a rate unconstrained by the current cwnd.

Reviewer 2: Does this look better?

According to {{!RFC4341}}, a DCCP CCID specifying TCP-like behavior is allowed to grow the cwnd without limit during an uncongested period when it sends at a rate unconstrained by the current cwnd.

Just wanted to make this part clear "grow an unlimited cwnd".

Also changed behaviour to behavior to make it uniform across the document.

Author: ```suggestion
According to {{!RFC4341}}, a DCCP CCID specifying TCP-like behavior is allowed to grow the cwnd without limit during an uncongested period when it sends at a rate unconstrained by the current cwnd.
```

> The text in {{Section 5.1 of !RFC4341}} is updated by this document by adding the text in ({{rules}}) to specify the management of the

Reviewer 2: This document updates {{Section 5.1 of !RFC4341}} by adding the text in {{rules}} to specify how the cwnd is managed when the sender is rate-limited.

Author: ```suggestion
This document updates {{Section 5.1 of !RFC4341}} by adding the text in {{rules}} to specify how the cwnd is managed when the sender is rate-limited.
```

> cwnd when the sender is rate-limited.

Author: ```suggestion
```

## RFC 5681: TCP Congestion Control

> {{!RFC5681}} specifies no cwnd increase limitation in the standard TCP behavior

Reviewer 2: {{!RFC5681}} specifies no limit on cwnd growth in the standard TCP behavior when a TCP sender is unable to send at the maximum rate allowed by the cwnd.

Author: ```suggestion
{{!RFC5681}} specifies no limit on cwnd growth in the standard TCP behavior
```

when a TCP sender is unable to send at the maximum rate allowed by the cwnd.

> {{!RFC5681}} is updated by this document by adding the text in ({{rules}}) to specify the management of the

Reviewer 2: This document updates {{!RFC5681}} by adding the text in {{rules}} to specify how the cwnd is managed when the sender is rate-limited.

Author: ```suggestion
This document updates {{!RFC5681}} by adding the text in {{rules}} to specify how the cwnd is managed when the sender is rate-limited.
```

cwnd when the sender is rate-limited.

> ## RFC 9002: The QUIC Transport Protocol

Reviewer 2: RFC 9002: QUIC Loss Detection and Congestion Control

Author: ```suggestion
## RFC 9002: QUIC Loss Detection and Congestion Control
```

{{Section 7.8 of !RFC9002}} states:

> >When bytes in flight is smaller than the congestion window and sending is not pacing limited, the congestion window is underutilized. This can happen due to insufficient application data or flow control limits. When this occurs, the congestion window SHOULD NOT be increased in either slow start or congestion avoidance.

Reviewer: ```suggestion
>"When bytes in flight is smaller than the congestion window and sending is not pacing limited, the congestion window is underutilized. This can happen due to insufficient application data or flow control limits. When this occurs, the congestion window SHOULD NOT be increased in either slow start or congestion avoidance."
```

This limits the cwnd growth in accordance with Rate-Limited Increase, but it is more conservative.

> {{!RFC9002}} is updated by this document by adding the text in ({{rules}}) to specify the management of the

Reviewer: ```suggestion
{{!RFC9002}} is updated by this document by replacing the final line in the cited text, by the text in in ({{rules}}) to specify the management of the
```

> {{!RFC9002}} is updated by this document by replacing the final line in the cited text with the text in ({{rules}}) to specify the management of the

Reviewer: ```suggestion
{{!RFC9002}} is updated by this document by replacing the final sentence in the cited text with the text in ({{rules}}) to specify the management of the
```

> {{!RFC9002}} is updated by this document by replacing the final line in the cited text, by the text in in ({{rules}}) to specify the management of the

Author: ```suggestion
{{!RFC9002}} is updated by this document by replacing the final line in the cited text with the text in ({{rules}}) to specify the management of the
```

> {{!RFC9002}} is updated by this document by replacing the final sentence in the cited text with the text in ({{rules}}) to specify the management of the

Reviewer 2: This document updates {{!RFC9002}} by replacing the final sentence of the cited text with the text in {{rules}} to specify how the cwnd is managed when the sender is rate-limited.

Author: ```suggestion
This document updates {{!RFC9002}} by replacing the final sentence of the cited text with the text in {{rules}} to specify how the cwnd is managed when the sender is rate-limited.
```

cwnd when the sender is rate-limited.

> ## RFC 9260: The Stream Control Transmission Protocol (SCTP)

Reviewer 2: RFC 9260: Stream Control Transmission Protocol

Author: ```suggestion
## RFC 9260: Stream Control Transmission Protocol
```

The quoted statement from {{!RFC9260}} limits the cwnd growth in accordance with Rate-Limited Increase, but it is more conservative. {{Section 7.2.1 of !RFC9260}} is limited to Slow Start. Congestion Avoidance is discussed in {{Section 7.2.2 of !RFC9260}} -- however, this section does not contain a similar rule. It is implicitly clear that the quoted statement from {{!RFC9260}} only applies to Slow Start.

{{!RFC9260}} is updated by this document by replacing the  the final sentence of quoted text with the text in ({{rules}}) to specify the management of the cwnd when the sender is rate-limited.

{{Section 7.2.2 of !RFC9260}} is also updated to add text in ({{rules}}).

This ensures that the update applies to both Slow Start and Congestion Avoidance.

## RFC 9438: CUBIC

{{Section 5.8 of !RFC9438}} says:

>"Cubic doesn't increase cwnd when it's limited by the sending application or rwnd".

This limits the cwnd growth in accordance with Rate-Limited Increase, but it
is more conservative.

{{!RFC9438}} is updated by this document by replacing this  text with the text in ({{rules}}) to specify the management of the

> cwnd when the sender is rate-limited.

Author: ```suggestion
```

is not in Fast Recovery.
Only when these two conditions are met can the cwnd be increased; otherwise, the cwnd MUST NOT be increased.

> The quoted statement from {{!RFC9260}} limits the cwnd growth in accordance with Rate-Limited Increase, but it is more conservative. {{Section 7.2.1 of !RFC9260}} is limited to Slow Start. Congestion Avoidance is discussed in {{Section 7.2.2 of !RFC9260}} -- however, this section does not contain a similar rule. It is thus implicitly clear that the quoted statement from {{!RFC9260}} only applies to Slow Start, whereas Rate-Limited Increase applies to both Slow Start and Congestion Avoidance.

Reviewer: ```suggestion
The quoted statement from {{!RFC9260}} limits the cwnd growth in accordance with Rate-Limited Increase, but it is more conservative. {{Section 7.2.1 of !RFC9260}} is limited to Slow Start. Congestion Avoidance is discussed in {{Section 7.2.2 of !RFC9260}} -- however, this section does not contain a similar rule. It is implicitly clear that the quoted statement from {{!RFC9260}} only applies to Slow Start.
```

> {{!RFC9260}} is updated by this document by adding the text in ({{rules}}) to specify the management of the

Reviewer: ```suggestion
{{!RFC9260}} is updated by this document by replacing the  the final line of quoted text with the text in ({{rules}}) to specify the management of the cwnd when the sender is rate-limited.

{{Section 7.2.2 of !RFC9260}} is also updated to add text in ({{rules}}). 

```

> cwnd when the sender is rate-limited.

Reviewer: ```suggestion
This ensures that the update applies to both Slow Start and Congestion Avoidance.
```

## RFC 9438: CUBIC

{{Section 5.8 of !RFC9438}} says:

> >Cubic doesn't increase cwnd when it's limited by the sending application or rwnd.

Reviewer: ```suggestion
>"Cubic doesn't increase cwnd when it's limited by the sending application or rwnd".
```

This limits the cwnd growth in accordance with Rate-Limited Increase, but it
is more conservative.

> {{!RFC9438}} is updated by this document by adding the text in ({{rules}}) to specify the management of the

Reviewer: ```suggestion
{{!RFC9438}} is updated by this document by replacing this  text with the text in ({{rules}}) to specify the management of the
```

is not in Fast Recovery.
Only when these two conditions are met can the cwnd be increased; otherwise, the cwnd MUST NOT be increased.

> The quoted statement from {{!RFC9260}} limits the cwnd growth in accordance with Rate-Limited Increase, but it is more conservative. {{Section 7.2.1 of !RFC9260}} is limited to Slow Start. Congestion Avoidance is discussed in {{Section 7.2.2 of !RFC9260}} -- however, this section does not contain a similar rule. It is implicitly clear that the quoted statement from {{!RFC9260}} only applies to Slow Start.

Reviewer 2: Just changed a couple of words:

The quoted statement from {{!RFC9260}} limits cwnd growth in accordance with Rate-Limited Increase, but it is more conservative. {{Section 7.2.1 of !RFC9260}} is limited to Slow Start. Congestion Avoidance is discussed in {{Section 7.2.2 of !RFC9260}}; however, this section does not contain a similar rule. It is therefore clear that the quoted statement from {{!RFC9260}} only applies to Slow Start.

Author: ```suggestion
The quoted statement from {{!RFC9260}} limits cwnd growth in accordance with Rate-Limited Increase, but it is more conservative. {{Section 7.2.1 of !RFC9260}} only discusses Slow Start. Congestion Avoidance is discussed in {{Section 7.2.2 of !RFC9260}}; however, this section does not contain a similar rule. It is therefore clear that the quoted statement from {{!RFC9260}} only applies to Slow Start.
```

> {{!RFC9260}} is updated by this document by replacing the  the final line of quoted text with the text in ({{rules}}) to specify the management of the cwnd when the sender is rate-limited.

Reviewer: ```suggestion
{{!RFC9260}} is updated by this document by replacing the  the final sentence of quoted text with the text in ({{rules}}) to specify the management of the cwnd when the sender is rate-limited.
```

> {{!RFC9260}} is updated by this document by replacing the  the final sentence of quoted text with the text in ({{rules}}) to specify the management of the cwnd when the sender is rate-limited.

Reviewer 2: This document updates {{!RFC9260}} by replacing the final sentence of the quoted text with the text in {{rules}} to specify how the cwnd is managed when the sender is rate-limited.

Note: even if this suggestion is avoided, we have to fix "the the" in the original sentence

Author: ```suggestion
This document updates {{!RFC9260}} by replacing the final sentence of the quoted text with the text in {{rules}} to specify how the cwnd is managed when the sender is rate-limited.
```

{{Section 7.2.2 of !RFC9260}} is also updated to add text in ({{rules}}).

This ensures that the update applies to both Slow Start and Congestion Avoidance.

## RFC 9438: CUBIC

{{Section 5.8 of !RFC9438}} says:

>"Cubic doesn't increase cwnd when it's limited by the sending application or rwnd".

This limits the cwnd growth in accordance with Rate-Limited Increase, but it
is more conservative.

> {{!RFC9438}} is updated by this document by replacing this  text with the text in ({{rules}}) to specify the management of the

Reviewer 2: This document updates {{!RFC9438}} by replacing this text with the text in {{rules}} to specify how the cwnd is managed when the sender is rate-limited.

Author: ```suggestion
This document updates {{!RFC9438}} by replacing this text with the text in {{rules}} to specify how the cwnd is managed when the sender is rate-limited.
```
